presto-product-tests/src/main/java/io/prestosql/tests/hive/TestRowByRowDelete.java

```diff
@@ -0,0 +1,264 @@
+        onHive().executeQuery(format("INSERT INTO %s VALUES (1, 100), (2, 200), (3, 300), (4, 400), (5, 500)", tableName));
+        String where = " WHERE column1 >= 2 AND column2 <= 400";
+        log.info("About to perform Presto delete of row %s", where);
+        onPresto().executeQuery(format("DELETE FROM %s %s", tableName, where));
+        log.info("Querying to verify that the Presto delete worked");
+        assertThat(onPresto().executeQuery(format("SELECT COUNT(*) FROM %S WHERE column1 IN (1, 5)", tableName)))
+                .containsOnly(row(2));
+        onPresto().executeQuery("DROP TABLE " + tableName);
+    }
+
+    @Test(groups = HIVE_TRANSACTIONAL)
+    public void testCreateOnHivePartitionAndRowsDelete()
+    {
+        onHive().executeQuery("CREATE TABLE create_on_hive_partition_and_rows_delete" +
+                " (column2 BIGINT) PARTITIONED BY (column1 INT) STORED AS ORC TBLPROPERTIES('transactional'='true')");
+        testPartitionAndRowsDelete("create_on_hive_partition_and_rows_delete");
+    }
+
+    @Test(groups = HIVE_TRANSACTIONAL)
+    public void testCreateOnPrestoPartitionAndRowsDelete()
+    {
+        onPresto().executeQuery("CREATE TABLE create_on_presto_partition_and_rows_delete" +
+                " (column2 BIGINT, column1 INT) WITH (transactional = true, partitioned_by = ARRAY['column1'])");
+        testPartitionAndRowsDelete("create_on_presto_partition_and_rows_delete");
+    }
+
+    // Test cases with both a partition to drop and rows to delete
+    private void testPartitionAndRowsDelete(String tableName)
+    {
+        log.info("About to insert rows");
+        onHive().executeQuery(format("INSERT INTO %s (column1, column2) VALUES (1, 100), (1, 200), (2, 300), (2, 400), (2, 500)", tableName));
+        String where = " WHERE column1 = 2 OR column2 = 200";
+        log.info("About to delete of rows %s", where);
+        onPresto().executeQuery(format("DELETE FROM %s %s", tableName, where));
+        assertThat(onPresto().executeQuery(format("SELECT column2 FROM %s", tableName)))
+                .containsOnly(row(100));
+        onPresto().executeQuery("DROP TABLE " + tableName);
+    }
+
+    private String makeInsertValues(int col1Value, int col2First, int col2Last)
+    {
+        checkArgument(col2First <= col2Last, "The first value %s must be less or equal to the last %s", col2First, col2Last);
+        return IntStream.rangeClosed(col2First, col2Last).mapToObj(i -> format("(%s, %s)", col1Value, i)).collect(Collectors.joining(", "));
+    }
+
+    // A method used only during IDE test debugging to keep the docker containers alive
+    private void testUninterruptibly(Runnable runnable)
+    {
+        try {
+            runnable.run();
+        }
+        catch (Throwable e) {
+            log.error("Saw exception %s", e, e);
+            Uninterruptibles.sleepUninterruptibly(Duration.ofDays(1));
+        }
+        Uninterruptibles.sleepUninterruptibly(Duration.ofDays(1));
+    }
```

Comment:
You should not need this at all

1. see https://github.com/prestosql/presto/pull/4996 / https://github.com/prestosql/presto/issues/4911
2. you can also do a bit more complicated way, but hopefully with shorter round-trip times
   - `presto-product-tests-launcher/bin/run-launcher env up --environment singlenode-hdp3 --debug --without-presto`
   - add your laptop's "external" IP (non `127.*`) to `/etc/hosts` as `hadoop-master` hostname, eg
      ```
      192.168.1.181	hadoop-master`
      ```
   - run Presto from IDE (using `presto-server-main/etc` configs + https://github.com/prestosql/presto/pull/5100)
   - run tests from IntelliJ directly